packages/singleton-manager/README.md

```diff
@@ -0,0 +1,297 @@
+# Singleton Manager
+
+A singleton manager provides a way to make sure a singleton instance loaded from multiple file locations stays a singleton.
+Primarily useful if two major version of a package with a singleton is used.
+
+## How to use
+
+### Installation
+
+```sh
+npm i --save singleton-manager
+```
+
+⚠️ You need to make SURE that only ONE version of `singleton-manager` is installed. For how see [Non Goals](#non-goals).
+
+### Example Singleton Users
+
+Use the same singleton for both versions (as we don't use any of the breaking features)
+
+```js
+// managed-my-singleton.js
+import { singletonManager } from 'singleton-manager';
+import { mySingleton } from 'my-singleton'; // is available as 1.x and 2.x via node resolution
+
+singletonManager.set('my-singleton/index.js::1.x', mySingleton);
+singletonManager.set('my-singleton/index.js::2.x', mySingleton);
+```
+
+OR create a special compatible version of the singleton
+
+```js
+// managed-my-singleton.js
+import { singletonManager } from 'singleton-manager';
+import { MySingleton } from 'my-singleton'; // is available as 1.x and 2.x via node resolution
+
+class CompatibleSingleton extends MySingleton {
+  // add forward or backward compatibility code
+}
+const compatibleSingleton = new CompatibleSingleton();
+
+singletonManager.set('my-singleton/index.js::1.x', compatibleSingleton);
+singletonManager.set('my-singleton/index.js::2.x', compatibleSingleton);
+```
+
+AND in you App then you need to load the above code BEFORE loading the singleton or any feature using it.
+
+```js
+import './managed-my-singleton.js';
+
+import { mySingleton } from 'my-singleton'; // will no always be what is "defined" in managed-my-singleton.js
+```
+
+### Warning
+
+Overriding version is an App level concern hence components of "features" are not allowed to use it.
+If you try to call multiple times for the same key then it will throw an error.
+
+```js
+// on app level
+singletonManager.set('my-singleton/index.js::1.x', compatibleSingleton);
+
+// somewhere in a dependency
+singletonManager.set('my-singleton/index.js::1.x', compatibleSingleton);
+
+// will throw an error that it is already defined
+```
+
+### Example Singleton Maintainers
```

Comment:
```suggestion
### Example Singleton Manager
```

Reply:
it's an example for people who create and maintain a singleton

Reply:
Wouldn't it still be more clear then to write:
```md
### Example Singleton Manager

If you are a maintainer of a singleton.... etc.
```
?

Reply:
added more info 👍